### PR TITLE
Added versioning

### PR DESCRIPTION
**Pull request by vbuterin**

### CODE REVIEW

File: specs/casper_sharding_v2.1.md [hunk 1]
```diff
@@ -455,7 +460,7 @@ For each one of these attestations:
...
 * Verify that `len(attester_bitfield) == ceil_div8(len(attestation_indices))`, where `ceil_div8 = (x + 7) // 8`. Verify that bits `len(attestation_indices)....` and higher, if present (i.e. `len(attestation_indices)` is not a multiple of 8), are all zero
 * Derive a group public key by adding the public keys of all of the attesters in `attestation_indices` for whom the corresponding bit in `attester_bitfield` (the ith bit is `(attester_bitfield[i // 8] >> (7 - (i %8))) % 2`) equals 1
-* Verify that `aggregate_sig` verifies using the group pubkey generated and `hash(slot.to_bytes(8, 'big') + parent_hashes + shard_id + shard_block_hash + justified_slot.to_bytes(8, 'big'))` as the message.
+* Let `version = pre_fork_version if slot < fork_slot_number else post_fork_version`. Verify that `aggregate_sig` verifies using the group pubkey generated and `hash(slot.to_bytes(8, 'big') + parent_hashes + shard_id + shard_block_hash + justified_slot.to_bytes(8, 'big'))` as the message.
```
Comment (djrtwo) [inline specs/casper_sharding_v2.1.md]: Need to add version to the msg

Comment (vbuterin) [reply specs/casper_sharding_v2.1.md]: Sorry! Just fixed.

**Comment by djrtwo** [comment]
@vbuterin There has been a discussion around defining the attestation message as an SSZ datastructure so we can remove any ambiguities in the generation of this message (big vs little endian, data type sizes, etc).

I think this is a good idea.

Thoughts?

**Comment by vbuterin** [comment]
Sounds good to me.

**Comment by vbuterin** [comment]
Done: https://github.com/ethereum/eth2.0-specs/pull/35/files#diff-6440c21b70e311de83d387cc32b028e3

**Comment by djrtwo** [comment]
One more correction @vbuterin

**Comment by hwwhww** [comment]
link: #5

**Comment by djrtwo** [comment]
Should be `attestation.justified_slot`. 
I'm going to fix and handle merge conflict

**Pull request merged**